crates/fuel-core/src/schema/balance.rs

```diff
@@ -79,8 +79,6 @@ impl BalanceQuery {
         Ok(balance)
     }
 
-    // TODO: This API should be migrated to the indexer for better support and
-    //  discontinued within fuel-core.
     #[graphql(complexity = "query_costs().balance_query")]
```

Comment:
```suggestion
    #[graphql(complexity = "query_costs().balance_query + query_costs().storage_iterator \
        + (query_costs().storage_read + first.unwrap_or_default() as usize) * child_complexity \
        + (query_costs().storage_read + last.unwrap_or_default() as usize) * child_complexity\")]
```

We need to limit the number of requested balances=)

Reply:
It also makes this change breaking

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2490/commits/fd2b0191c7ede59eec061bc7f5d78c2de60ef0ab